### PR TITLE
feat: In dns parse, when the parameter `resolvers` is empty, the local dns is used as `resolvers` by default 

### LINKED ISSUE #2422: feat: In dns parse, when the parameter `resolvers` is empty, will use the default `resolvers`
### Issue description

In `dns parse`. When we use `dns parse` in the plugin, the parameter `resolvers` is empty, then we need to use local `dns` as resolvers by default.

### PR DESCRIPTION
**Pull request by Firstsawyou**
fix #2422

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible?

### CODE REVIEW

File: t/core/utils.t [hunk 1]
```diff
@@ -68,3 +68,19 @@ qr/random seed \d+\ntwice: false/
 GET /t
 --- no_error_log
 [error]
+
```
Comment (moonming) [inline t/core/utils.t:71]: The test case did not cover empty configuration of resolver

**Review by moonming — CHANGES_REQUESTED** [review]

File: t/core/utils.t (hunk 1 above)
Comment (Firstsawyou) [reply t/core/utils.t:71]: added.

Comment (moonming) [reply t/core/utils.t:71]: No, I means the `config.yaml` file

Comment (Firstsawyou) [reply t/core/utils.t:71]: Here we do not need to set the `dns_resolver` in `config.yaml`. Through the `set_resolver` function in the ʻutils` file. We can set `resolvers` to be empty for testing.

Comment (Firstsawyou) [reply t/core/utils.t:71]: Here we do not need to set the `dns_resolver`  in `config.yaml`. Through the `set_resolver` function in the `utils`  file. We can set `resolvers` to be empty for testing.

File: apisix/core/utils.lua [hunk 2]
```diff
@@ -100,11 +101,16 @@ local function dns_parse(resolvers, domain)
         return nil, "unsupport DNS answer"
     end
 
-    return dns_parse(resolvers, answer.cname)
+    return dns_parse(answer.cname, resolvers)
```
Comment (membphis) [inline apisix/core/utils.lua:104]: I think we can remove the argument `resolvers` here

File: apisix/init.lua [hunk 3]
```diff
@@ -177,7 +178,7 @@ end
 
 
 local function parse_domain(host)
-    local ip_info, err = core.utils.dns_parse(dns_resolver, host)
+    local ip_info, err = core.utils.dns_parse(host, dns_resolver)
```
Comment (membphis) [inline apisix/init.lua:181]: ditto

File: t/core/utils.t [hunk 4]
```diff
@@ -68,3 +68,50 @@ qr/random seed \d+\ntwice: false/
...
+
+=== TEST 3: resolvers is not empty
+--- config
+    location /t {
+        content_by_lua_block {
+            local core = require("apisix.core")
+            local resolvers = {"8.8.8.8"}
+            core.utils.set_resolver(resolvers)
+            local ip_info, err = core.utils.dns_parse("baidu.com", resolvers)
```
Comment (membphis) [inline t/core/utils.t:81]: please change it to `github.com`

File: t/core/utils.t [hunk 5]
```diff
@@ -68,3 +68,50 @@ qr/random seed \d+\ntwice: false/
 GET /t
 --- no_error_log
 [error]
+
+
+
+=== TEST 3: resolvers is not empty
```
Comment (membphis) [inline t/core/utils.t:74]: bad title

File: t/core/utils.t [hunk 6]
```diff
@@ -68,3 +68,50 @@ qr/random seed \d+\ntwice: false/
...
+--- request
+GET /t
+--- response_body eval
+qr/"address":.+,"name":"baidu.com"/
+--- no_error_log
+[error]
+--- LAST
+
+
+=== TEST 4: resolvers is empty
```
Comment (membphis) [inline t/core/utils.t:97]: bad title

File: apisix/init.lua (hunk 3 above)
Comment (Firstsawyou) [reply apisix/init.lua]: removed.

File: t/core/utils.t (hunk 6 above)
Comment (Firstsawyou) [reply t/core/utils.t]: already changed.

File: t/core/utils.t (hunk 4 above)
Comment (Firstsawyou) [reply t/core/utils.t]: already changed.

**Review by membphis — APPROVED** [review]

**Comment by Firstsawyou** [comment]
@moonming ping

**Review by moonming — APPROVED** [review]

**Pull request merged**